### ISSUE TITLE
Integrating Focal loss from TFaddons

### ISSUE THREAD

**Issue by AakashKumarNain** [tensorflow/models#8418]
**Describe the feature you request and the current behavior/state.**
Focal loss has been [implemented](https://github.com/tensorflow/models/blob/3b13794fdb762517ad7c90361b6a47b4f46acc76/official/vision/detection/modeling/losses.py#L25) and used in the RetinaNet model in the `model garden`. There is an existing implementation of [focal loss](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/losses/focal_loss.py) in `addons` as well. 

Having multiple implementations for the same functionality creates redundancy in the ecosystem which creates confusion for the end users. In order to get rid of this redundancy, we propose to use the focal loss from addons. 
 
**Challenges**
None. The code has been tested on CPU/GPU but not on TPU as we don't have a way to test functionlities on TPU as of now but that shouldn't be a problem because all the ops used are TPU compatible as well. 

**Are you willing to contribute it (Yes/No)?** Yes

**Comment by saberkun** [comment]
Adding @xianzhidu  to track and review this.

**Comment by pengchongjin** [comment]
Detection models under official/ is currently undergoing major refactor. We will reuse the standard implementation and release it later.

**Comment by saberkun** [comment]
Closing this issue for now. The internal version of tensorflow addons is stale and TF/Keras team does not have effort to keep that up-to-date.
